lib/task_listener.js

```diff
@@ -113,26 +110,51 @@ export default class TaskListener extends EventEmitter {
     return cancelListener;
   }
 
-  get availableCapacity() {
+  async availableCapacity() {
     // Note: Sometimes capacity could be zero (dynamic capacity changes based on
     // shutdown and other factors) so subtracting runningTasks could result in a
     // negative number, hence the use of at least returning 0.
-    return Math.max(this.capacity - this.runningTasks.length, 0);
+    let deviceCapacity;
+    try {
+      deviceCapacity = await this.deviceManager.getAvailableCapacity();
+    }
+    catch (e) {
+      // If device capacity ccannot be determined for device managers configured
+      // for the worker, then default to 0
+      debug('Error determining device capacity');
+      deviceCapacity = 0;
+    }
+
+    let runningCapacity = Math.max(this.capacity - this.runningTasks.length, 0);
+    let hostCapacity = Math.min(runningCapacity, deviceCapacity);
+
+    if (hostCapacity < runningCapacity) {
+      this.runtime.log('[info] host capacity adjusted',
+        {
+          message: `The available running capacity of the host has been changed.` +
+                   ` Available Capacities: Device: ${deviceCapacity} ` +
+                   `Running: ${runningCapacity} Adjusted Host Capacity: ${hostCapacity}`
+        }
+      );
+    }
+
+    return hostCapacity;
   }
 
   async getTasks() {
-    if (this.availableCapacity === 0)  return;
+    let availableCapacity = await this.availableCapacity();
```

Comment:
Can we reliably call bitbar every 5s... If not maybe the phone manager needs some sort of rate limiter to make sure that it doesn't actually call more than say once a minute...